decisions/2019/10/09-fe-decision-board.md

```diff
@@ -0,0 +1,226 @@
+handled.
+
+Will run the trial for 1 week in **cli-style** before rolling out to
+**ui-widgets**. If everything keeps going all right we will identify
+another candidate from the [Dependabot enabled
+repos](https://app.dependabot.com/accounts/dhis2).
+
+#### Defer the automatic release system
+
+There are a few options of varying complexity:
+
+- The proposed workflow in the agenda.
+
+- Bundle multiple changes on the same branch before merging to `master`.
+
+- Using `[skip ci]` in a commit message to, well, skip CI, which will
+  also skip the `semantic-release` process.
+
+- Implementing another commit message tag, e.g. `[defer release]` to
+  still run the CI checks, but skip the `semantic-release` process. [A PR
+  for it has been posted](https://github.com/dhis2/cli/pull/161).
+
+The `[skip ci]` and `[defer release]` options are different variations
+of the same, so we will go ahead with one of them. If the defer-release
+PR is accepted, we will use that. If not, we will use skip-ci.
+
+#### Use the same workflow everywhere
+
+Currently we use "squash-merge" as our default strategy, and we also use
+`next` branches to merge multiple fixes/features to `master` at the same
+time.
+
+There is a significant downside of this in terms of how the changelogs
+are generated.
+
+This commit was merged from `next` to `master` to release version 3.0.0
+of `@dhis2/ui-core`:
+https://github.com/dhis2/ui-core/commit/6df182bdbb1de8b80d76ac7677a1a37478aa8da6
+
+The commit message itself is nicely formatted and has a list of all the
+changes, but the CHANGELOG is quite sparse:
+https://github.com/dhis2/ui-core/blob/master/CHANGELOG.md#300-2019-06-06
+
+It contains a single `chore` and lists the breaking change. In order to
+get all the commits into the changelog, they must land on `master` as
+well. We can use the `[skip ci]`/`[defer release]` tag to do that when
+there are two or three PRs to merge after eachother, but when there are
+[69 commits bundled into one
+release](https://github.com/dhis2/ui-core/pull/430/commits) that is not
+an option.
+
+Given that we want:
+
+- Linear history on `master` (no merge commits)
+- One commit per feature/fix
+- Option for bundling changes for a release
+- Precise changelogs
+
+Then the "rebase-merge" strategy is a good option:
+
+- We can enforce that _all_ commits on a branch are semantic commits as
+  as status check (using **probot-semantic-pull-requests** currently enabled
+  in all repositories).
+
+- No need to ever provide a manual commit message in GitHub when
+  merging a PR.
+
+- "rebase-merge" is not a true rebase on GitHub, it results in a
+  fast-forward merge just like "squash-merge".
+
+- If a PR contains multiple commits, they will land as-is on `master`
+  and show up in the changelog correctly, so we can have a `fix` and
+  `feat` in the same PR.
+
+- If a PR contains a single commit, it will also land as-is on `master`.
+
+- It is up to the developer to decide if a PR should be squashed before
```

Comment:
A disadvantage with this is that you lose the individual commits forever. Before you could always go back to the closed PR (eg. from the reference of the squashed commit in github)  and see the individual commits before they were squashed together.

Reply:
I've added your concerns to the "known issues" section.